shared/middleware/helpers/bucket_helper.rb

```diff
@@ -346,6 +343,13 @@ def list_versions(encrypted_channel_id, filename)
     track_list_operation 'BucketHelper.list_versions'
     s3.list_object_versions(bucket: @bucket, prefix: key).
       versions.
+      # Sort descending order by time stamp.
+      sort {|a, b| b.last_modified <=> a.last_modified}.
```

Comment:
slight variation (just a thought)
```suggestion
      sort_by(&:last_modified).reverse.
```